File: app/src/main/bilbil_api/dto/ChatRoomInfoResponse.java
package com.longtoast.bilbil_api.dto;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class ChatRoomInfoResponse {

    private Integer roomId;

    private ItemDTO item;

    private UserDTO lender;
    private UserDTO borrower;

    @Data
    @Builder
    public static class ItemDTO {
        private Long id;
        private String title;
        private Integer price;
        private String imageUrl;
    }

    @Data
    @Builder
    public static class UserDTO {
        private Integer id;
        private String nickname;
        private String profileImageUrl;
    }
}

File: app/src/main/java/com/longtoast/bilbil/dto/ChatRoomInfoResponse.kt
package com.longtoast.bilbil.dto

data class ChatRoomInfoResponse(
    val roomId: Int,
    val item: ItemDTO,
    val lender: UserDTO,
    val borrower: UserDTO
) {
    data class ItemDTO(
        val id: Int,
        val title: String,
        val price: Int,
        val imageUrl: String?
    )

    data class UserDTO(
        val id: Int,
        val nickname: String?,
        val profileImageUrl: String?
    )
}
